confine backup and restore breaking changes to the version they were introduced

conceptualshark opened the pull request:
## Description

Messaging around breaking changes has been introduced periodically into the SM backup and restore docs, and then duplicated when new versions are cut. This leads to confusions around when a breaking change applies. 

This PR confines breaking change messaging to the versions they were introduced. This PR does _not_ include any messaging around 8.7/ in /next, as that messaging should be added when the new backup and restore procedures and changes are complete.

Breaking change notifications should remain in the operate/tasklist doc for 8.6 for the new changes introduced there.

_Internal slack ask: https://camunda.slack.com/archives/C026U8GBNSW/p1736376448383119_

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

conceptualshark commented:
Just a note that I'm unsure who to tag specifically for an engineering review, so I've dropped an ask with the Zeebe team to take a look.

PHWaechtler reviewed (changes requested):
For Optimize, this change was introduced with 8.2. Lets remove the notice from all >8.2 for Optimize (for the other components I do not know which is the correct version). See [slack thread](https://camunda.slack.com/archives/C55U06YRG/p1736862887716389)

PHWaechtler reviewed (dismissed):
Lgtm for Optimize

akeller commented:
Adding the EMs as reviewers for Operate and Tasklist.

ThorbenLindhauer reviewed (dismissed):
Removing the paragraphs as in the diff makes a lot of sense to me.

I'd like to suggest to add a section to the 8.6 section of the Announcements page, where we put this content centrally: https://docs.camunda.io/docs/reference/announcements/#camunda-86

My reasoning is that people will continue updating Camunda from 8.5 to 8.6 and they will confuse which docs version they are on, so it's even easier to miss this change now (it was already before, because it was listed not on the central migration page).

Btw I believe this section has the wrong heading level: https://docs.camunda.io/docs/reference/announcements/#zeebe-spring-sdk-test-support-in-867

edit: you can also take my review on behalf of @marcosgvieira, I think we don't need to have his review to merge this.

conceptualshark commented:
@ThorbenLindhauer I've added this to both the 8.6 announcement pages for visibility (And fixed that heading's indentation!), if you are able to review again!

ThorbenLindhauer commented:
@conceptualshark Thank you. We should also add the changes to the backup endpoints there, or? And for the default actuator ports, let's mention the components that are affected by this.

conceptualshark commented:
@ThorbenLindhauer @akeller  Do you think the older changes introduced in 8.2 are worth placing into the [historical announcements](https://docs.camunda.io/docs/8.3/reference/announcements/#camunda-82)? There's no centralized place for 8.2 docs any longer, so that's the only place they could be listed.

Can you confirm that Operate and Tasklist are the only components impacted by the actuator port change? Thank you!

ThorbenLindhauer commented:
> Do you think the older changes introduced in 8.2 are worth placing into the [historical announcements](https://docs.camunda.io/docs/8.3/reference/announcements/#camunda-82)? There's no centralized place for 8.2 docs any longer, so that's the only place they could be listed.

Sounds fine to me.

> Can you confirm that Operate and Tasklist are the only components impacted by the actuator port change?

I don't know. We can either just document what we are seeing here or ask around in the different component teams.

conceptualshark commented:
This should now:

- Remove the warning changing `backup` -> `backups` in all versions, and the warning about backup API changes, as this was introduced in 8.2. A note has been added to the 8.2 announcements in versions 8.3 - 8.6. 
- Remove the actuator port change from 8.8 (leaving it in 8.6, where it was introduced, and 8.7, as a warning for upgrading). A note has been added to the 8.6 announcements.

conceptualshark commented:
@akeller This should be up-to-date now based on what we talked about yesterday - it might be easiest to review in the preview environment, but should remove the extra/duplicated version warnings throughout.

akeller reviewed (approved):
I am happy with this 👍
